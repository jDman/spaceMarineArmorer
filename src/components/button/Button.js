import React, { Fragment } from 'react';

const Button = (props) => {
  const type = props.type || 'button';

  const classString = Array.isArray(props.classes)
    ? props.classes.join(' ')
    : typeof props.classes === 'string'
    ? props.classes
    : '';

  return (
    <Fragment>
      <button
        type={type}
        className={classString}
        disabled={props.disabled || props.loading}
        onClick={(event) =>
          props.clickHandler ? props.clickHandler(event) : () => {}
        }
      >
        {props.buttonText}
      </button>
    </Fragment>
  );
};

export default Button;
