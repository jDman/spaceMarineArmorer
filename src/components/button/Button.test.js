import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button />', () => {
  it('should render with all props included', () => {
    const props = {
      type: 'button',
      buttonText: 'test button',
      classes: ['test', 'classes'],
      disabled: false,
      loading: false,
      clickHandler: () => {},
    };
    const component = renderer.create(
      <Button
        type={props.type}
        classes={props.classes}
        disabled={props.disabled}
        loading={props.loading}
        clickHandler={props.clickHandler}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render without unnecessary props', () => {
    const props = {
      clickHandler: () => {},
    };
    const component = renderer.create(
      <Button clickHandler={props.clickHandler} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
