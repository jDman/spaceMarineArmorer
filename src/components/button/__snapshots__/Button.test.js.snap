// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render with all props included 1`] = `
<button
  className="test classes"
  disabled={false}
  onClick={[Function]}
  type="button"
/>
`;

exports[`<Button /> should render without unnecessary props 1`] = `
<button
  className=""
  onClick={[Function]}
  type="button"
/>
`;
